centreon/www/include/monitoring/recurrentDowntime/formDowntime.php

```diff
@@ -40,6 +40,60 @@
     exit();
 }
 
+/**
+ * GetUserAclAllowedResources returns allowed resources for the user regarding resource type
+ *
+ * @param CentreonACL $userAcl
+ * @param string $resourceType
+ */
+function getUserAclAllowedResources(CentreonACL $userAcl, string $resourceType)
+{
+    return match ($resourceType) {
+        'hosts' => $userAcl->getHostAclConf(null, 'broker'),
+        'servicegroups' => $userAcl->getServiceGroupAclConf(null, 'broker'),
+        'hostgroups' => $userAcl->getHostGroupAclConf(null, 'broker'),
+    };
+}
+
+/**
+ * Check resources access regarding selected resources
+ *
+ * @param CentreonACL $userAcl
+ * @param int[] $selectedResources
+ * @param string $resourceType
+ * @return bool
+ **/
+function checkResourcesRelations(CentreonACL $userAcl, array $selectedResources, string $resourceType): bool
+{
+    $allowedResources = getUserAclAllowedResources($userAcl, $resourceType);
+
+    $selectedResourceIds = array_map(
+        function (array $resource) {
+            return $resource['id'];
+        },
+        $selectedResources
+    );
+
+    $diff = array_diff($selectedResourceIds, array_keys($allowedResources));
+
+    if (empty($diff)) {
```

Comment:
Equivalent suggestion for relative readability.
```suggestion
    if ([] === $diff) {
```

Reply:
As Laurent Calvet said, we should **BAN** the empty() 👍 
I'll change this